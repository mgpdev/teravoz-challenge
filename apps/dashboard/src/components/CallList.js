import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import List from "@material-ui/core/List";
import moment from "moment";
import { subscribeToQueue } from "./../services/web-socket";
import { formatPhoneNumber } from "./../utils/formatter";
import logger from "./../utils/logger";
import CallItem from "./CallItem";

const styles = theme => ({
  root: {
    width: "100%",
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper
  }
});

class CallList extends React.Component {
  constructor(props) {
    super(props);
    this.state = { calls: [] };
    subscribeToQueue(props.queue, calls => {
      logger(`Event (${props.queue}) Received`, calls);
      this.setState({
        calls: JSON.parse(calls)
      });
    });
  }
  render() {
    const { title, classes } = this.props;
    return (
      <div className={classes.root}>
        <Typography variant="title" gutterBottom>
          {title}
        </Typography>
        <List>
          {this.state.calls.map(row => {
            return (
              <CallItem
                number={formatPhoneNumber(row.number)}
                time={moment(row.startAt).fromNow()}
              />
            );
          })}
        </List>
      </div>
    );
  }
}

CallList.propTypes = {
  title: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(CallList);
