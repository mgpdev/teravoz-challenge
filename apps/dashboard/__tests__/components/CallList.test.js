import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

// Component to be tested
import CallList from "./../../src/components/CallList";

describe("<CallList />", () => {
  describe("render()", () => {
    test("renders the component", () => {
      const wrapper = shallow(
        <CallList title="New customers calls" queue="900" />
      );
      const component = wrapper.html();

      expect(component).toMatchSnapshot();
    });
  });
});
