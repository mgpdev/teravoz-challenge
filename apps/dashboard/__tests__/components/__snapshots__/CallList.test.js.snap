// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CallList /> render() renders the component 1`] = `"<div class=\\"CallList-root-1\\"><h2 class=\\"MuiTypography-root-2 MuiTypography-title-8 MuiTypography-gutterBottom-19\\">New customers calls</h2><ul class=\\"MuiList-root-27 MuiList-padding-28\\"><div><li class=\\"MuiListItem-root-31 MuiListItem-default-34 MuiListItem-gutters-38\\"><div class=\\"MuiAvatar-root-42 MuiAvatar-colorDefault-43\\"><svg class=\\"MuiSvgIcon-root-45\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><g><path d=\\"M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z\\"></path></g></svg></div><div class=\\"MuiListItemText-root-52\\"><span class=\\"MuiTypography-root-2 MuiTypography-subheading-9 MuiListItemText-primary-55\\">(11) 9986-5645</span><p class=\\"MuiTypography-root-2 MuiTypography-body1-11 MuiTypography-colorTextSecondary-25 MuiListItemText-secondary-56\\">11:50</p></div></li></div><div><li class=\\"MuiListItem-root-31 MuiListItem-default-34 MuiListItem-gutters-38\\"><div class=\\"MuiAvatar-root-42 MuiAvatar-colorDefault-43\\"><svg class=\\"MuiSvgIcon-root-45\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><g><path d=\\"M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z\\"></path></g></svg></div><div class=\\"MuiListItemText-root-52\\"><span class=\\"MuiTypography-root-2 MuiTypography-subheading-9 MuiListItemText-primary-55\\">(11) 9986-5645</span><p class=\\"MuiTypography-root-2 MuiTypography-body1-11 MuiTypography-colorTextSecondary-25 MuiListItemText-secondary-56\\">11:50</p></div></li></div><div><li class=\\"MuiListItem-root-31 MuiListItem-default-34 MuiListItem-gutters-38\\"><div class=\\"MuiAvatar-root-42 MuiAvatar-colorDefault-43\\"><svg class=\\"MuiSvgIcon-root-45\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><g><path d=\\"M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z\\"></path></g></svg></div><div class=\\"MuiListItemText-root-52\\"><span class=\\"MuiTypography-root-2 MuiTypography-subheading-9 MuiListItemText-primary-55\\">(11) 9986-5645</span><p class=\\"MuiTypography-root-2 MuiTypography-body1-11 MuiTypography-colorTextSecondary-25 MuiListItemText-secondary-56\\">11:50</p></div></li></div></ul></div>"`;
